web-common/src/lib/analytics/posthog.ts

```diff
@@ -0,0 +1,43 @@
+import posthog, { type Properties } from "posthog-js";
+
+const POSTHOG_API_KEY = import.meta.env.RILL_UI_PUBLIC_POSTHOG_API_KEY;
+
+export function initPosthog(rillVersion: string, sessionId?: string | null) {
+  // No need to proceed if PostHog is already initialized
+  if (posthog.__loaded) return;
+
+  if (!POSTHOG_API_KEY) {
+    console.warn("PostHog API Key not found");
+    return;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+  posthog.init(POSTHOG_API_KEY, {
+    api_host: "https://us.i.posthog.com", // TODO: use a reverse proxy https://posthog.com/docs/advanced/proxy
+    session_recording: {
```

Comment:
Is it possible to configure to not upload when on slow network? I wouldnt want an app I use to upload videos when I am using mobile network for example.

Reply:
We could consider disabling it for slow connections, but I'm not sure we should. It's important for us to understand user behavior on all connection types (e.g. users on slow networks might quit in frustration if we don't have proper loading states). The PostHog SDK is designed to be efficient, e.g. it batches & compresses events to minimize bandwidth. That said, if we see significant issues, we could add a network condition check in the future.